### DISCUSSION TITLE
Walkthrough for setting up Android Studio to build jidoujisho apk

### DISCUSSION THREAD

**Discussion by ShiroiKuma0** [arianneorpilla/jidoujisho#285] [Show and tell]
Hi:

Would anybody here be willing to guide me through the proper setup for Android Studio - so that I could build a jidoujisho apk on my PC. I don't have enough experience with AS to know exactly how to go about building a Flutter project in AS.

I want to experience with some code modifications - but need to verify that my changes then are working.

However I'm not able to get even the clean jidoujisho apk to build from a github clone - so my reasoning is I'm not clear on how to build it.

I've experienced with simple AS projects before - however here basically:

- I've setup AS - latest
- installed flutter and dart plugins
- add local.properties file to yuuna/android directory specifying "flutter.sdk=" with the location of Flutter SDK
- then open the yuuna directory in AS as a project
- enable Dart for it when AS asks
- then close the project in AS
- open the yuuna/android directory as a project - it runs the background build tasks
- then I can run Build -- Build Bundle(s) / APK(s) -- Build APK(s), which fails with:

yuuna\.dart_tool\package_config.json does not exist.
Did you run this command from the same directory as your pubspec.yaml file?
Target gen_dart_plugin_registrant failed: Exception: 

So, I know I'm doing something wrong and don't understand the workflow. Could someone point me in the right direction - or show me where my process is flawed - so that I could build the apk?

Many thanks.

**Comment by arianneorpilla** [comment]
I wouldn't bother with Android Studio at all, I'd just use VS Code and a terminal like many Flutter developers

You really don't need to do most of what you've just described, I'd just set up Flutter with the tutorial you can find on the Flutter site and `flutter doctor` till everything is set up then just `flutter run`

**Reply by ShiroiKuma0** [reply]
Oh - thank you very much, didn't realize it's this simple - just ran 'flutter build apk' and it does build it, easy - for some reason thought it must be built in Android Studio. I can play with it now... :@)

**Reply by arianneorpilla** [reply]
`flutter build apk --split-per-abi`

**Comment by ShiroiKuma0** [comment]
Hello - I'm wondering if you can still give me a hand here...

I had to reinstall my PC - so had it setup and working before, however that was still with Android Studio installed, which installed the Android SDK.

Now, I can't get it to build - per: https://docs.flutter.dev/get-started/install/windows#android-setup
**Note: Flutter relies on a full installation of Android Studio to supply its Android platform dependencies. However, you can write your Flutter apps in a number of editors; a later step discusses that.**
So it seems to me you can't have flutter build the project without installing AS. Or am I wrong here.

Anyhow, I installed AS, to get the dependencies - now if I connect the phone to the PC, `flutter run` attempts to build the project for Android - except I don't think I did it that way (phone connected) before. And anyway it bombs with a whole slew of errors like:

E/AndroidRuntime(20211): Caused by: java.lang.IllegalArgumentException: Tried to schedule job for non-existent component: ComponentInfo{com.sec.android.diagmonagent/com.sec.android.diagmonagent.dma.selflog.ExceptionService}

If I try to run it without the phone connected - it attempts to build it for Windows - which is wrong - and requests Visual Studio install, which is not the way to go.

So how can I not worry about installing AS, and still have flutter build packages for Android? What am I missing here that I did before?

Many thanks for helping me with this if possible.

**Reply by arianneorpilla** [reply]
I would simply run a `flutter doctor` to see what could be wrong. Try a `flutter clean` as well, 

I honestly don't have much to go on other than what you've given me, check out the `structured-content-beta` branch of the project. I resolved a compile error related to building for Android in that branch but I haven't merged that into `main` yet.

**Reply by arianneorpilla** [reply]
I've merged the branch I've mentioned into `main`. You can pull the latest `main` from the repo and try building with that. Try using the latest Flutter stable.
